coordinator/core/src/main/kotlin/net/consensys/zkevm/ethereum/coordination/conflation/GlobalBlobAwareConflationCalculator.kt

```diff
@@ -32,10 +35,101 @@ class GlobalBlobAwareConflationCalculator(
   override val lastBlockNumber: ULong
     get() = conflationCalculator.lastBlockNumber
 
+  private val gasUsedInBlobHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BLOB,
+    name = "gas",
+    description = "Total gas in each blob"
+  )
+  private val compressedDataSizeInBlobHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BLOB,
+    name = "compressed.data.size",
+    description = "Compressed L2 data size in bytes of each blob"
+  )
+  private val uncompressedDataSizeInBlobHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BLOB,
+    name = "uncompressed.data.size",
+    description = "Uncompressed L2 data size in bytes of each blob"
+  )
+  private val gasUsedInBatchHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BATCH,
+    name = "gas",
+    description = "Total gas in each batch"
+  )
+  private val compressedDataSizeInBatchHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BATCH,
+    name = "compressed.data.size",
+    description = "Compressed L2 data size in bytes of each batch"
+  )
+  private val uncompressedDataSizeInBatchHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BATCH,
+    name = "uncompressed.data.size",
+    description = "Uncompressed L2 data size in bytes of each batch"
+  )
+  private val avgCompressedTxDataSizeInBatchHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BATCH,
+    name = "avg.compressed.tx.data.size",
+    description = "Average compressed transaction data size in bytes of each batch"
+  )
+  private val avgUncompressedTxDataSizeInBatchHistogram = metricsFacade.createHistogram(
+    category = LineaMetricsCategory.BATCH,
+    name = "avg.uncompressed.tx.data.size",
+    description = "Average uncompressed transaction data size in bytes of each batch"
+  )
+
   init {
     conflationCalculator.onConflatedBatch(this::handleBatchTrigger)
   }
 
+  private fun recordBatchMetrics(conflation: ConflationCalculationResult) {
+    runCatching {
+      val gasUsedInBatch = blobBlockCounters
+        .filter { conflation.blocksRange.contains(it.blockNumber) }
+        .sumOf { it.gasUsed }
+      val uncompressedDataSizeInBatch = blobBlockCounters
+        .filter { conflation.blocksRange.contains(it.blockNumber) }
+        .sumOf { it.blockRLPEncoded.size }
+      val compressedDataSizeInBatch = blobCalculator.getCompressedDataSizeInCurrentBatch()
+      val numOfTransactionsInBatch = blobBlockCounters
```

Comment:
Can we reuse the result of the filtering?

Reply:
Good catch, missed that one, thanks and updated